### DISCUSSION TITLE
Updated web title reverts to original value

### DISCUSSION THREAD

**Discussion by oliver-tbs** [pnp/pnpcore#1053] [Q&A]
Hi,

https://pnp.github.io/pnpcore/using-the-sdk/webs-intro.html#updating-a-web

I have implemented this code sample into an azure function app. The code executes without error and does change the web title, however, more often than not the title reverts to the original value upon page refresh of the SharePoint team site. Is there a  mechanism at play here that I need to account for?

Thank you!

**Comment by jansenbe** [comment]
@oliver-tbs : I don't see that happening in my testing @oliver-tbs . Can you verify the permissions you've used (e.g. Sites.FullControl), the authentication model (delegated or application permissions) and the type of site (connected to a Microsoft 365 Group or not).

**Reply by oliver-tbs** [reply]
Hi! I am using the following application level permissions:

Microsoft Graph: Team.Create, TeamMember.ReadWrite.All, TeamSettings.ReadWrite.All
SharePoint: Sites.FullControl.All

And the site is connected to a Microsoft 365 group.

**Reply by jansenbe** [reply]
Let me do some testing using application permissions and see if that makes a difference @oliver-tbs. Using delegated on a group connected  site worked fine for me

**Reply by jansenbe** [reply]
@oliver-tbs : also tested with application permissions and that worked fine as well. Do you see this on other sites/tenants as well?

**Reply by oliver-tbs** [reply]
Only the one tenant available and I have run the code against multiple sites. When I isolate the update code from my overall procedure and run it, I get better results: the new name sticks a while longer before reverting back. Knowing the issue is something on my end helps though. Thanks very much for your time looking into this. I will update if I figure out a solution.

**Reply by oliver-tbs** [reply]
An observation I've made is that the issue presents itself when using a queue trigger function but not when using an http trigger function.

**Reply by jansenbe** [reply]
@oliver-tbs : weird: might the code error out before the call succeeds? Maybe you're not using the correct permissions for the failing path?

**Reply by oliver-tbs** [reply]
After more observation it does not appear related to http vs. queue trigger. Rather I think it has something to do with the "order of operations" that I am trying to implement. Let me provide a higher level view:

- queue trigger function calls an internal function that uses MS Graph to create the team site:

                       var teamResponse = await graphClient.Teams.Request().AddResponseAsync(team);

- next I call another internal function that uses PnP.Core to update the web.Title property:

                       var web = await context.Web.GetAsync(p => p.Title);
                       web.Title = displayName;
                       await web.UpdateAsync();
                        
- once the queue function is finished and I load the new team site, I see the updated name
- then I do a ctrl-R refresh and it reverts to the original title

**Reply by jansenbe** [reply]
@oliver-tbs : Guess there's some timing issues and some part of the team site provisioning kicks in after you've already done your updates. You're creating a Teams team using Graph, the site will then be created asynchronously.
